lib/thermite/tasks.rb

```diff
@@ -109,7 +111,7 @@ def define_build_task
           FileUtils.cp(config.rust_path('target', target, config.shared_library),
                        config.ruby_path('lib'))
         elsif !download_binary
-          raise cargo_required_msg
+          raise cargo_required_msg unless options[:optional_build]
```

Comment:
In my opinion, `mkmf` output is for debugging purposes, whereas warnings and errors should be printed to `stderr`.